src/backend/distributed/utils/jsonbutils.c

```diff
@@ -0,0 +1,120 @@
+#include "postgres.h"
+
+#include "pg_version_compat.h"
+
+#include "catalog/namespace.h"
+#include "catalog/pg_class.h"
+#include "catalog/pg_collation.h"
+#include "catalog/pg_type.h"
+
+#include "utils/array.h"
+#include "utils/json.h"
+#include "distributed/jsonbutils.h"
+#include "distributed/metadata_cache.h"
+
+#include "utils/builtins.h"
+#include "utils/lsyscache.h"
+#include "fmgr.h"
+
+
+/*
+ * ExtractFieldJsonb gets value of fieldName from jsonbDoc and puts it
+ * into result. If not found, returns false. Otherwise, returns true.
+ * The field is returned as a Text* Datum if as_text is true, or a Jsonb*
+ * Datum if as_text is false.
+ */
+static bool
+ExtractFieldJsonb(Datum jsonbDoc, const char *fieldName, Datum *result, bool as_text)
+{
+	Datum pathArray[1] = { CStringGetTextDatum(fieldName) };
+	bool pathNulls[1] = { false };
+	bool typeByValue = false;
+	char typeAlignment = 0;
+	int16 typeLength = 0;
+	int dimensions[1] = { 1 };
+	int lowerbounds[1] = { 1 };
+
+	get_typlenbyvalalign(TEXTOID, &typeLength, &typeByValue, &typeAlignment);
+
+	ArrayType *pathArrayObject = construct_md_array(pathArray, pathNulls, 1, dimensions,
+													lowerbounds, TEXTOID, typeLength,
+													typeByValue, typeAlignment);
+	Datum pathDatum = PointerGetDatum(pathArrayObject);
+
+	/*
+	 * We need to check whether the result of jsonb_extract_path is NULL or not, so use
+	 * FunctionCallInvoke() instead of other function call api.
+	 *
+	 * We cannot use jsonb_path_exists to ensure not-null since it is not available in
+	 * postgres 11.
```

Comment:
we don't use PG 11 anymore, maybe good time to remove useless code

Reply:
I just removed the comments here.
Using a different function call api does not really reduce the code size so there is no benefit I can see to rewriting this portion of the code.

(I tried rewriting with directfunctioncall macro)